common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -98,32 +98,28 @@ static ap_wlan_hal::Event dwpal_to_bwl_event(const std::string &opcode)
 
 static uint8_t dwpal_bw_to_beerocks_bw(const uint8_t chan_width)
 {
-
-    // 0 => CHAN_WIDTH_20_NOHT
-    // 1 => CHAN_WIDTH_20
-    // 2 => CHAN_WIDTH_40
-    // 3 => CHAN_WIDTH_80
-    // 4 => CHAN_WIDTH_80P80
-    // 5 => CHAN_WIDTH_160
-    // 6 => CHAN_WIDTH_UNKNOWN
-    uint8_t bw = chan_width;
-
-    if (chan_width > 6) {
+    /*
+             DWPAL structure    eqv     eWiFiBandwidth
+        0 => CHAN_WIDTH_20_NOHT  | 1 => BANDWIDTH_20
+        1 => CHAN_WIDTH_20       | 1 => BANDWIDTH_20
+        2 => CHAN_WIDTH_40       | 2 => BANDWIDTH_40
+        3 => CHAN_WIDTH_80       | 3 => BANDWIDTH_80
+        4 => CHAN_WIDTH_80P80    | 4 => BANDWIDTH_80_80
+        5 => CHAN_WIDTH_160      | 5 => BANDWIDTH_160
+        6 => CHAN_WIDTH_UNKNOWN  | 0 => BANDWIDTH_UNKNOWN
+    */
+
+    if (chan_width == 6) /*CHAN_WIDTH_UNKNOWN*/ {
         LOG(ERROR) << "Invalid bandwidth value: " << int(chan_width);
-        return 0;
+        return int(eWiFiBandwidth::BANDWIDTH_UNKNOWN);
     }
-
-    // Convert to beerocks value (we have only one value for 20Mhz bandwidth)
-    if (chan_width > 0) {
-        bw -= 1;
-    }
-
-    // Treat 80P80 as 160
-    if (chan_width >= 3) {
-        bw = (uint8_t)BANDWIDTH_160;
+    if (chan_width == 0) /*CHAN_WIDTH_20_NOHT*/ {
+        //Currently prplMesh does not support No HT
```

Comment:
I'm not sure about **doesn't support** , IMO we just handle both 20 && 20_NOHT the same.

Reply:
I agree. We support it, we just do not distinguish between it and regular 20Mhz bandwidth.
So I would change it to:
`// prplMesh does not distinguish between 'CHAN_WIDTH_20_NOHT' and 'CHAN_WIDTH_20'.`

Reply:
Changed